pkg/parse/run.go

```diff
@@ -280,63 +301,98 @@ func read(ctx context.Context, p Parser, trigger string, state *reconcilerState,
 	return status.Append(sourceStatus.errs, setSourceStatusErr)
 }
 
-// readFromSource reads the source or hydrated configs, checks whether the sourceState in
-// the cache is up-to-date. If the cache is not up-to-date, reads all the source or hydrated files.
-// readFromSource returns the rendering status and source status.
-func readFromSource(ctx context.Context, p Parser, trigger string, state *reconcilerState, sourceState sourceState) (renderingStatus, sourceStatus) {
-	opts := p.options()
-	start := time.Now()
-
-	hydrationStatus := renderingStatus{
-		commit: sourceState.commit,
-	}
-	sourceStatus := sourceStatus{
-		commit: sourceState.commit,
-	}
-
+// readHydrationStatus reads from the file path which the hydration-controller
+// container writes to. It checks if the hydrated files are ready and returns
+// a renderingStatus.
+func readHydrationStatus(p Parser, srcState sourceState, hydrationStatus renderingStatus) (sourceState, renderingStatus) {
+	options := p.options()
 	// Check if the hydratedRoot directory exists.
-	// If exists, read the hydrated directory. Otherwise, read the source directory.
-	absHydratedRoot, err := cmpath.AbsoluteOS(opts.HydratedRoot)
+	// If exists, read the hydrated directory. Otherwise, fail.
+	absHydratedRoot, err := cmpath.AbsoluteOS(options.HydratedRoot)
 	if err != nil {
 		hydrationStatus.message = RenderingFailed
 		hydrationStatus.errs = status.InternalHydrationError(err, "hydrated-dir must be an absolute path")
-		return hydrationStatus, sourceStatus
+		return srcState, hydrationStatus
 	}
 
 	var hydrationErr hydrate.HydrationError
 	if _, err := os.Stat(absHydratedRoot.OSPath()); err == nil {
-		sourceState, hydrationErr = opts.readHydratedDir(absHydratedRoot, opts.reconcilerName, sourceState)
+		srcState, hydrationErr = options.readHydratedDir(absHydratedRoot, options.reconcilerName, srcState)
 		if hydrationErr != nil {
 			hydrationStatus.message = RenderingFailed
 			hydrationStatus.errs = status.HydrationError(hydrationErr.Code(), hydrationErr)
-			return hydrationStatus, sourceStatus
+			return srcState, hydrationStatus
 		}
 		hydrationStatus.message = RenderingSucceeded
 	} else if !os.IsNotExist(err) {
 		hydrationStatus.message = RenderingFailed
 		hydrationStatus.errs = status.InternalHydrationError(err, "unable to evaluate the hydrated path %s", absHydratedRoot.OSPath())
-		return hydrationStatus, sourceStatus
+		return srcState, hydrationStatus
+	} else {
+		// Source of truth does not require hydration, but hydration-controller is running
+		hydrationStatus.message = RenderingNotRequired
+		hydrationStatus.requiresRendering = false
+		err := hydrate.NewTransientError(fmt.Errorf("rendering enabled but not required"))
```

Comment:
nit: Use RenderingNotRequired?

Reply:
Yeah I'm not really sure what is best to populate here, as normally we insert an actual `error` that was returned from a function into the status. In this case, it's a contrived error because the error condition is defined by us. It seems a bit redundant to use the same string in `message` and `errors` but I don't currently have a better alternative

Reply:
I updated the error to provide a bit more technical detail about the error, while the message is intended to remain a bit more user-friendly. PTAL